src/datepicker/datepicker.js

```diff
@@ -990,3 +997,9 @@ function(scope, element, attrs, $compile, $parse, $document, $rootScope, $positi
     }
   };
 });
+
+function clearWatchListeners(watchListeners) {
```

Comment:
This function should not be outside the controllers of relevance - this will leak as a global function.

In this case, the functionality is simple enough to not require a helper function. Something like
```js
while (watchListeners.length) {
  watchListeners.shift()();
}
```
in all relevant areas should be enough